go/cmd/dolt/commands/push.go

```diff
@@ -85,95 +95,236 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
+			cli.Println("hint: Updates were rejected because the tip of your current branch is behind")
+			cli.Println("hint: its remote counterpart. Integrate the remote changes (e.g.")
+			cli.Println("hint: 'dolt pull ...') before pushing again.")
 		}
-		return HandleVErrAndExitCode(verr, usage)
+		return err
 	}
 
-	remoteDB, err := opts.Remote.GetRemoteDB(ctx, dEnv.DoltDB.ValueReadWriter().Format(), dEnv)
+	postPushRemoteBranches, err := getRemoteBranches(queryist, sqlCtx)
 	if err != nil {
-		err = actions.HandleInitRemoteStorageClientErr(opts.Remote.Name, opts.Remote.Url, err)
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		return fmt.Errorf("failed to get post-push remote branches: %w", err)
 	}
 
-	tmpDir, err := dEnv.TempTableFilesDir()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	changesMade := getChangesMade(remoteBranches, postPushRemoteBranches)
+	if !changesMade {
+		cli.Println("Everything up-to-date")
 	}
-	var verr errhand.VerboseError
-	err = actions.DoPush(ctx, dEnv.RepoStateReader(), dEnv.RepoStateWriter(), dEnv.DoltDB, remoteDB, tmpDir, opts, buildProgStarter(defaultLanguage), stopProgFuncs)
+
+	return nil
+}
+
+func getChangesMade(remoteBranches map[string]remoteBranchInfo, postPushRemoteBranches map[string]remoteBranchInfo) bool {
+	changesMade := false
+	if len(remoteBranches) != len(postPushRemoteBranches) {
+		changesMade = true
+	} else {
+		for name, remoteBranch := range remoteBranches {
+			updatedRemoteBranch, ok := postPushRemoteBranches[name]
+			if !ok {
+				changesMade = true
+				break
+			}
+			if remoteBranch.Hash != updatedRemoteBranch.Hash {
+				changesMade = true
+				break
+			}
+		}
+	}
+	return changesMade
+}
+
+func getRemotes(queryist cli.Queryist, sqlCtx *sql.Context) (map[string]remoteInfo, error) {
+	rows, err := GetRowsForSql(queryist, sqlCtx, "select * from dolt_remotes")
 	if err != nil {
-		verr = printInfoForPushError(err, opts.Remote, opts.DestRef, opts.RemoteRef)
+		return nil, fmt.Errorf("failed to read dolt remotes: %w", err)
 	}
 
-	if opts.SetUpstream {
-		err := dEnv.RepoState.Save(dEnv.FS)
+	remotes := map[string]remoteInfo{}
+	for _, row := range rows {
+		name := row[0].(string)
+		url := row[1].(string)
+
+		fetchSpecsJson, err := getJsonDocumentCol(sqlCtx, row[2])
 		if err != nil {
-			err = fmt.Errorf("%w; %s", actions.ErrFailedToSaveRepoState, err.Error())
+			return nil, fmt.Errorf("failed to read fetch specs for remote %s: %w", name, err)
+		}
+		fetchSpecsArray, ok := fetchSpecsJson.Val.([]interface{})
+		if !ok {
+			return nil, fmt.Errorf("failed to read fetch specs for remote %s", name)
+		}
+		fetchSpecs := []string{}
+		for _, spec := range fetchSpecsArray {
+			text, ok := spec.(string)
+			if !ok {
+				return nil, fmt.Errorf("failed to read fetch specs for remote %s: %w", name, err)
+			}
+			fetchSpecs = append(fetchSpecs, text)
+		}
+
+		paramsJson, err := getJsonDocumentCol(sqlCtx, row[3])
```

Comment:
It's not clear to me what we support in this arbitrary hash map of parameters. There seems to be a lack of documentation on what this structure allows, which is not really a problem for this PR.

Reply:
Yeah, clear lack of docs. Since params aren't really supported (by any of the existing dolt code), we can remove the Params property and only add it back when we add params support. Thoughts?